include/seqan3/range/container/aligned_sequence_adaptor_constant_access.hpp

```diff
@@ -0,0 +1,688 @@
+    };
+
+    //!\brief Construct by range given by two iterators of an aligned sequence.
+    constexpr aligned_sequence_adaptor_constant_access(iterator it1, iterator it2)
+    {
+        assign(it1, it2);
+    };
+
+    //!\brief Construct by sequence.
+    constexpr aligned_sequence_adaptor_constant_access(inner_type sequence_)
+    {
+        assign(sequence_);
+    };
+
+    //!\brief Construct by initializer_list for inner sequence container.
+    constexpr aligned_sequence_adaptor_constant_access(std::initializer_list<value_type> l)
+    {
+        assign(l);
+    };
+    //!\}
+
+    /*!\name Iterators
+     * \{
+    */
+    //!\brief Return iterator pointing to first element of underlying sequence.
+    auto begin() noexcept
+    {
+        return iterator{*this, 0};
+    }
+
+    //!\brief Return iterator pointing to past-the-end element of gapped sequence.
+    auto end() noexcept
+    {
+        return iterator{*this, size()};
+    }
+
+    //!\brief Return const iterator pointing to first element of underlying sequence
+    const_iterator cbegin() const noexcept
+    {
+        return const_iterator{*this, 0};
+    }
+
+    //!\brief Return const iterator pointing to past-the-end element of underlying sequence.
+    const_iterator cend() const noexcept
+    {
+        return const_iterator{*this, size()};
+    }
+     //!\}
+
+    /*!\name Boolean operators
+     * \{
+     */
+    /*!\brief Equality operator for aligned sequences.
+    *
+    * Two aligned sequences are the same if their literal sequences and gap
+    * positions are the same.
+    */
+    bool operator==(aligned_sequence_t const & rhs) const
+    {
+        return sequence == rhs.sequence && gap_vector == rhs.gap_vector;
+    }
+
+    //!\brief Unequality operator for aligned sequences.
+    bool operator!=(aligned_sequence_t const & rhs) const
+    {
+        return !(*this == rhs);
+    }
+
+    //!\brief Swap two aligned sequences and their support structures.
+    void swap(aligned_sequence_t & rhs)
+    {
+        sequence.swap(rhs.sequence);
+        std::swap(gap_vector, rhs.gap_vector);
+        rs.set_vector(&gap_vector);
+        rhs.rs.set_vector(&rhs.gap_vector);
+        ss.set_vector(&gap_vector);
+        rhs.ss.set_vector(&gap_vector);
+    }
+
+    //!\brief Return gapped sequence length.
+    size_type size() const noexcept
+    {
+        return gap_vector.size();
+    }
+
+    /*!\brief Return the maximal aligned sequence length.
+    *
+    * The maximal sequence length is limited by either the maximal size of the
+    * compressed sequence or the gap vector.
+    */
+    size_type max_size() const
+    {
+        return std::min<size_type>(sequence.max_size(), gap_vector.max_size());
+    }
+
+    //!\brief An aligned sequence is empty if it contains no alphabet letters or gaps.
+    bool empty() const
+    {
+        return sequence.empty() && gap_vector.size() == 0;
+    }
+    //!\}
+
+    /*!\name Assignment
+     * \{
```

Comment:
For consistency I would prefer if you used the same "groups" for you members as seqan3::concatenated_sequences does (which in turn is modeled on http://en.cppreference.com/w/cpp/container/vector), ie.
* Member types
* Constructors, destructor and assignment
* Iterators
* Element access
* Capacity
* Modifiers
* Comparison operators